cluster.go

```diff
@@ -80,6 +82,9 @@ type ClusterOptions struct {
 	IdleCheckFrequency time.Duration
 
 	TLSConfig *tls.Config
+
+	// Hooks for telemetry
+	IncrLogHook func(name string, log string, err error)
```

Comment:
IncrLogHook will increment the metric `name` and log the shared `log` line with the provided `err` (if any).